build_script.py

```diff
@@ -112,7 +118,7 @@ def main():
     run("{0} -c {1} -emit-object {2} -module-name XCTest -parse-as-library -emit-module "
         "-emit-module-path {3}/XCTest.swiftmodule -o {3}/XCTest.o -force-single-frontend-invocation "
         "-module-link-name XCTest".format(swiftc, style_options, " ".join(sourcePaths), build_dir))
-    run("clang {0}/XCTest.o -shared -o {0}/libXCTest.so -Wl,--no-undefined -Wl,-soname,libXCTest.so -L{1}/lib/swift/linux/ -lswiftGlibc -lswiftCore -lm".format(build_dir, swift_build_dir))
+    run("clang {1}/lib/swift/linux/{2}/swift_begin.o {0}/XCTest.o {1}/lib/swift/linux/{2}/swift_end.o -shared -o {0}/libXCTest.so -Wl,--no-undefined -Wl,-soname,libXCTest.so -L{1}/lib/swift/linux/ -lswiftGlibc -lswiftCore -lm".format(build_dir, swift_build_dir, arch))
```

Comment:
Fantastic. Thanks, @hpux735! :100: 

Personally I've been meaning to reformat this convoluted string interpolation to use identifiers--something like the following:

```python
run('clang {swift_build_dir}/lib/swift/linux/{arch}/swift_begin.o '
    '{build_dir}/XCTest.o '
    '{swift_build_dir}/lib/swift/linux/{arch}/swift_end.o '
    '-shared -o {build_dir}/libXCTest.so '
    '-Wl,--no-undefined -Wl,-soname,libXCTest.so '
    '-L{swift_build_dir}/lib/swift/linux/ '
    '-lswiftGlibc -lswiftCore -lm'.format(
        build_dir=build_dir,
        swift_build_dir=swift_build_dir,
        arch=arch))
```

But I can take care of that at a later time. I think this is fine for now!